api/src/indigo_loaders.cpp

```diff
@@ -52,6 +89,35 @@ IndigoSdfLoader::~IndigoSdfLoader()
 {
 }
 
+IndigoObject* IndigoSdfLoader::next()
+{
+    if (sdf_loader->isEOF())
+        return 0;
+    
+    int counter = sdf_loader->currentNumber();
+    long long offset = sdf_loader->tell();
+    
+    sdf_loader->readNext();
+    
+    return new IndigoRdfMolecule(sdf_loader->data, sdf_loader->properties, counter, offset);
+}
+
+IndigoObject* IndigoSdfLoader::at(int index)
+{
+    sdf_loader->readAt(index);
+    return new IndigoRdfMolecule(sdf_loader->data, sdf_loader->properties, index, 0LL);
+}
+
+bool IndigoSdfLoader::hasNext()
+{
+    return !sdf_loader->isEOF();
+}
+
+long long IndigoSdfLoader::tell()
+{
+    return sdf_loader->tell();
+}
+

```

Comment:
Indeed these methods of `IndigoSdfLoader` were not implemented? 
Anyway, this is a fix and it does not have relation to the subject of the pull request (support Json for Ketcher). These changes should be extracted to a separate pull request and issue.